src/providers/execution/contracts/CSModule.py

```diff
@@ -1,72 +1,92 @@
 from web3.contract.contract import Contract, ContractEvent
 from web3.types import BlockIdentifier, EventData
 
 from src import variables
+from src.web3py.extensions.lido_validators import NodeOperatorId
 
 logger = logging.getLogger(__name__)
 
 
+# TODO: Move to types?
+class NodeOperatorSummary(NamedTuple):
+    targetLimitMode: int
+    targetValidatorsCount: int
+    stuckValidatorsCount: int
+    refundedValidatorsCount: int
+    stuckPenaltyEndTimestamp: int
+    totalExitedValidators: int
+    totalDepositedValidators: int
+    depositableValidatorsCount: int
+
+
 class CSModule(Contract):
     abi_path = "./assets/CSModule.json"
 
-    MAX_OPERATORS_COUNT = 2 ** 64
+    MAX_OPERATORS_COUNT = 2**64
 
     # TODO: Inherit from the base class.
     def __init__(self, address: ChecksumAddress | None = None) -> None:
         with open(self.abi_path, encoding="utf-8") as f:
             self.abi = json.load(f)
         super().__init__(address)
 
-    def get_stuck_node_operators(self, l_block: BlockIdentifier, r_block: BlockIdentifier) -> Iterable:
+    # TODO: Consider adding cache.
+    def current_stuck_operators_ids(self, block: BlockIdentifier = "latest") -> Iterable[NodeOperatorId]:
+        # TODO: Check performance on a large amount of node operators in a module.
+        if self.is_deployed(block):
+            for no_id in self.node_operators_ids(block):
+                if self.node_operator_summary(no_id, block).stuckValidatorsCount > 0:
+                    yield no_id
+
+    def frame_stuck_operators_ids(self, l_block: BlockIdentifier, r_block: BlockIdentifier) -> Iterable[NodeOperatorId]:
         """Returns node operators assumed to be stuck for the given frame (defined by the blockstamps)"""
 
         l_block_number = self.w3.eth.get_block(l_block).get("number", BlockNumber(0))
         r_block_number = self.w3.eth.get_block(r_block).get("number", BlockNumber(0))
         assert l_block_number <= r_block_number
 
-        by_no_id: Callable[[EventData], int] = lambda e: int(
-            e["args"]["nodeOperatorId"]
-        )  # TODO: Maybe it's int already?
+        by_no_id: Callable[[EventData], int] = lambda e: e["args"]["nodeOperatorId"]
+        by_block: Callable[[EventData], int] = lambda e: e["blockNumber"]
 
-        events = sorted(self.get_stuck_keys_events(r_block_number), key=by_no_id)
+        events = sorted(self.get_stuck_keys_events(l_block_number, r_block_number), key=by_no_id)
         for no_id, group in groupby(events, key=by_no_id):
-            last_event = sorted(tuple(group), key=lambda e: e["blockNumber"])[-1]
+            last_event = sorted(tuple(group), key=by_block)[-1]
             # Operators unstucked at the very beginning of the frame are fine.
             if (
-                last_event["args"]["stuckValidatorsCount"] == 0 and
+                last_event["args"]["stuckKeysCount"] == 0 and
                 last_event["blockNumber"] <= l_block_number
```

Comment:
`last_event["blockNumber"] == l_block_number`
should be enough, I guess

Reply:
btw the case is so rare that I think we don't need this branch in code at all

Reply:
No, we have to check for the case. But, I've changed the flow a little, so this check is swallowed here.